code/datums/wires/airlock.dm

```diff
@@ -38,50 +36,55 @@
-			A.loseMainPower()
-		if(WIRE_BACKUP1, WIRE_BACKUP2) // Pulse to loose backup power.
-			A.loseBackupPower()
-		if(WIRE_OPEN) // Pulse to open door (only works not emagged and ID wire is cut or no access is required).
-			if(A.obj_flags & EMAGGED)
-				return
-			if(!A.requiresID() || A.check_access(null))
-				if(A.density)
-					INVOKE_ASYNC(A, /obj/machinery/door/airlock.proc/open)
+	if(A.hasPower()) //Multitool has no effect at all if the door has lost power
+		switch(wire)
+			if(WIRE_POWER1, WIRE_POWER2) // Pulse to loose power.
+				A.loseMainPower()
+			if(WIRE_BACKUP1, WIRE_BACKUP2) // Pulse to loose backup power.
+				A.loseBackupPower()
+			if(WIRE_OPEN) // Pulse to open door (only works not emagged and ID wire is cut or no access is required).
+				if(A.obj_flags & EMAGGED)
+					return
+				if(!A.requiresID() || A.check_access(null))
+					if(A.density)
+						INVOKE_ASYNC(A, /obj/machinery/door/airlock.proc/open)
+					else
+						INVOKE_ASYNC(A, /obj/machinery/door/airlock.proc/close)
+			if(WIRE_BOLTS) // Pulse to toggle bolts
+				if(!A.locked)
+					A.bolt()
 				else
-					INVOKE_ASYNC(A, /obj/machinery/door/airlock.proc/close)
-		if(WIRE_BOLTS) // Pulse to toggle bolts (but only raise if power is on).
-			if(!A.locked)
-				A.bolt()
-			else
-				if(A.hasPower())
 					A.unbolt()
-			A.update_icon()
-		if(WIRE_IDSCAN) // Pulse to disable emergency access and flash red lights.
-			if(A.hasPower() && A.density)
-				A.do_animate("deny")
-				if(A.emergency)
-					A.emergency = FALSE
-					A.update_icon()
-		if(WIRE_AI) // Pulse to disable WIRE_AI control for 10 ticks (follows same rules as cutting).
-			if(A.aiControlDisabled == 0)
-				A.aiControlDisabled = 1
-			else if(A.aiControlDisabled == -1)
-				A.aiControlDisabled = 2
-			addtimer(CALLBACK(A, /obj/machinery/door/airlock.proc/reset_ai_wire), 1 SECONDS)
-		if(WIRE_SHOCK) // Pulse to shock the door for 10 ticks.
-			if(!A.secondsElectrified)
-				A.set_electrified(MACHINE_DEFAULT_ELECTRIFY_TIME, usr)
-		if(WIRE_SAFETY)
-			A.safe = !A.safe
-			if(!A.density)
-				A.close()
-		if(WIRE_TIMING)
-			A.normalspeed = !A.normalspeed
-		if(WIRE_LIGHT)
-			A.lights = !A.lights
-			A.update_icon()
+				A.update_icon()
+			if(WIRE_IDSCAN) // Pulse to disable emergency access and flash red lights.
+				if(A.hasPower() && A.density)
+					A.do_animate("deny")
+					if(A.emergency)
+						A.emergency = FALSE
+						A.update_icon()
+			if(WIRE_AI) // Pulse to disable WIRE_AI control for 10 ticks (follows same rules as cutting).
+				if(A.aiControlDisabled == 0)
+					A.aiControlDisabled = 1
+				else if(A.aiControlDisabled == -1)
+					A.aiControlDisabled = 2
+				addtimer(CALLBACK(A, /obj/machinery/door/airlock.proc/reset_ai_wire), 1 SECONDS)
+			if(WIRE_SHOCK) // Pulse to shock the door for 10 ticks.
+				if(!A.secondsElectrified)
+					A.set_electrified(MACHINE_DEFAULT_ELECTRIFY_TIME, usr)
+				if(isliving(usr))
+					A.shock(usr, 100)
```

Comment:
Didn't this already shock you, since electrified shocks you if the UI is open

Reply:
My mistake on that one if so, verifying and fixing along with CE blueprint testing.